Setup quiz create route

giacoletti opened the pull request:
### This PR contains:
- generates quiz model with category, type, difficulty, question, correct_answer, incorrect_answers
- creates quiz model spec to test factory
- creates quiz_category model
- creates quiz_category model spec for factory
- creates request spec for quizzes create route
- creates quizzes_controller.rb
- installs webmock and requires it in rails_helper.rb
- adds before hook with stub_request in request spec for quiz create route
- creates empty_questions.json and history_questions.json fixtures
- adds create route for quizzes, making Trivia Api call and returning set of questions

[User story](https://www.pivotaltracker.com/n/projects/2545113/stories/180631928)

@gsim1903 @elvitak @Mljungho @DefyCab

giacoletti committed:
Generates quiz model with category, type, difficulty, question, correct_answer, incorrect_answers
creates quiz model spec to test factory

giacoletti committed:
Creates quiz_category model
creates quiz_category model spec for factory
creates request spec for quizzes create route
creates quizzes_controller.rb

tochman reviewed (changes requested):
Creating a route does not feel like a feature. What is the corresponding PT story for this PR. Please provide a link in the description.

elvitak committed:
add rest client
WIP addcreate method to quiz controller

tochman committed:
refactors table structure
finalizes Qizzes_controlle#create
modifies migrations
NOTE: your db nedds to be re-created

giacoletti committed:
Installs webmock and requires it in rails_helper.rb
adds before hook with stub_request in request spec for quiz create route
creates empty_questions.json and history_questions.json fixtures

giacoletti committed:
Adds category column to quizzes table
adds unit specs for quiz model to check table columns and model validations
adds quizzes.rb model validations for all fields
adds category param permit to quiz_params method in quizzes_controller.rb

giacoletti committed:
Adds request spec for quiz create unsuccessfully with missing params

giacoletti committed:
Adds validate_params_presence method in quizzes_controller.rb

giacoletti committed:
Adds request spec to quizzes create route unsuccessfully due to missing category and due to missing difficulty
adds error handling for category and difficulty params missing in create method in quizzes_controller.rb

giacoletti committed:
Fixes error render in case Quiz.create fails in create method in quizzes_controller.rb
